scripts/powershell/Verify-And-Merge-PRs.ps1

```diff
@@ -0,0 +1,145 @@
+param(
+    [Parameter(Mandatory = $true)]
+    $AuthToken,
+
+    [Parameter(Mandatory = $true)]
+    $PRDataArtifactPath
+)
+
+$PRObjs = [System.Collections.ArrayList]::new()
+$MergedPRs = [System.Collections.ArrayList]::new()
+$ClosedPRs = [System.Collections.ArrayList]::new()
+$UnexpectedState = [System.Collections.ArrayList]::new()
+$OpenAndCleanPRs = [System.Collections.ArrayList]::new()
+$OpenButDirtyPRs = [System.Collections.ArrayList]::new()
+
+
+$GitHubAPIUrlBase = "https://api.github.com/repos/"
+$Headers = @{
+    Authorization = "bearer $AuthToken"
+}
+
+function Merge-PRs()
+{
+    if (($OpenButDirtyPRs.Count -gt 0) -or ($ClosedPRs.Count -gt 0) -or ($UnexpectedState.Count -gt 0))
+    {
+        Write-Output "The following PRs are either closed, or in an Un-mergeable state"
+        Write-Output "If Closed, Please investigate why a Sync PR was closed prematurely."
+        Write-Output "Ensure all checks are green and pr is approved. Merge them manually or run the sync pipeline again"
+
+        foreach ($obj in $OpenButDirtyPRs)
```

Comment:
I would add a header independently for each of these categories (assuming count > 0) which states exactly what to do for each different type.